### ISSUE TITLE
Filter dropdown sometimes makes unnecessary queries to data source

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#16101]
**Describe the bug**
From https://github.com/metabase/metabase/pull/15943#pullrequestreview-654993149 - continuation of #13832
When a dashboard has a "A list of all values" filter, which has more than 10,000 characters of total distinct values (ex. `select length(string_agg(my_column, '')) from my_table`) are not present in `metabase_fieldvalues` (or if cached field values has never been scanned), then a query is being made to the data source to request 5000 results on every load.

This happens even if [scanning is disabled](https://www.metabase.com/docs/latest/administration-guide/01-managing-databases.html#choose-when-metabase-syncs-and-scans) in Admin > Databases > (db) > Scheduling-tab > select "Never, I'll do this manually if I need to" and save.

**Workaround** Admin > Data Model > (db) > (table) > (column) :gear: > change the "Filtering on this field" from "A list of all values" to either "Search box" or "Plain input box"

If you have a lot of tables/fields, then it might be easier to make the change in the application database.
Go to the application database table `metabase_field` and change the column `has_field_values` to `none` or `search`

**To Reproduce**
Using Sample Dataset, but on MariaDB/Postgres, so it's easier to see queries being executed.
1. Admin > Data Model > Sample Dataset > People > City > set as "A list of all values" (dropdown)
2. Simple question > Sample Dataset > People - save as "Q1" to dashboard
3. On dashboard add a category filter and connect to "Q1".City - save dashboard
4. Refresh the dashboard and notice query like this:
```
SELECT `PEOPLE`.`CITY` AS `CITY` FROM `PEOPLE` GROUP BY `PEOPLE`.`CITY` ORDER BY `PEOPLE`.`CITY` ASC LIMIT 5000
```
5. Same thing happens, when going to Simple question > Sample Dataset > People > click City column > Filter - it sends a query to the data source, but the results are not being used for anything, since it doesn't show a dropdown - likely related to #7603

**Expected behavior**
This should not happen. It's basically running the same query as scan does nightly on every dashboard load.

**Information about your Metabase Installation:**
Testing commit `aad24de` (almost 0.39.2)

Related to #16102

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by jeremyjh** [comment]
It would be great if there was one place we could go to completely disable all of these features across the board. We don't use the Metabase Query UI at all, and building a list of unique values in some of our audit and history tables was absolutely crushing performance on the server. I'm actually quite astonished that default behaviour includes building unique values lists of un-indexed fields.

**Comment by weiyu1zhu** [comment]
It might make more sense for me that if we are disabling scanning by selecting "Never, I'll do this manually if I need to", then no matter what the "Filtering on this field" is, no query would be executed. It seems that this one has higher priority than the database sync settings for now, which makes me confused.

**Comment by qnkhuat** [comment]
Fixed as of current master(c413a00d70f7f76376c4f3f30efeeca2053db013), I'm not sure when we fixed this but version 44 should not have this problem.
